internal/internal_worker.go

```diff
@@ -1370,6 +1376,9 @@ func (aw *WorkflowReplayer) replayWorkflowHistory(logger log.Logger, service wor
 			SdkMetadata:                     true,
 		},
 	}
+	if aw.disableDeadlockDetection {
+		params.DeadlockDetectionTimeout = math.MaxInt64
```

Comment:
```suggestion
		params.DeadlockDetectionTimeout = unlimitedDeadlockDetectionTimeout
```

Same thing, but we already have a constant for this. But not a merge blocker, just if you make another commit.